### PR TITLE
Display warning about loss displayed & LOG_RANK with PP enable.

### PR DESCRIPTION
**Pull request by lkhphuc**
I found this question yesterday #931 after puzzling a while if there's something wrong with my PP.
IMO this is not obvious a logging issue so I add a warning based on @lessw2020 comments for when PP is enabled.

### CODE REVIEW

File: torchtitan/distributed/parallel_dims.py [hunk 1]
```diff
@@ -28,6 +28,12 @@ class ParallelDims:
 
     def __post_init__(self):
         self._validate()
+        if self.pp_enabled:
```
Comment (tianyu-l) [inline torchtitan/distributed/parallel_dims.py:31]: I'm not sure if this is a good place to put this warning, as `ParallelDims` doesn't know what loss will be put on non-final PP ranks.

I don't know what's the best place to put this warning though, how about right after https://github.com/pytorch/torchtitan/blob/main/torchtitan/train.py#L92

Comment (lkhphuc) [reply torchtitan/distributed/parallel_dims.py:31]: I initially put it there but afraid there might be some concern about complicating the train.py file so I put it right when it is created.

> as ParallelDims doesn't know what loss will be put on non-final PP ranks

I'm not sure what this mean. Here the log only suggest users to switch LOG_RANK to the last PP rank, and which is it the user has to figure out based on their nodes' configuration right?

**Comment by wconstab** [comment]
I like the idea of adding a warning. I wonder if we can do it in a way that it only prints the warning on the ranks that do not have the loss. Though that might make train.py more complicated, since that's the most obvious place we know 'has_last_stage'.

File: torchtitan/distributed/parallel_dims.py (hunk 1 above)
Comment (H-Huang) [reply torchtitan/distributed/parallel_dims.py:31]: Maybe we can add a utility that checks the values of the env `LOG_RANK`. If the last_stage rank is not included in `LOG_RANK` then print the warning? Adding a separate util function for this logic and calling this in train.py does not seem that intrusive / complicated.

> Here the log only suggest users to switch LOG_RANK to the last PP rank, and which is it the user has to figure out based on their nodes' configuration right?

You can look at the pp_mesh as @tianyu-l  mentioned to figure out which is the last PP rank and in the warning let the users know which rank they should get LOG_RANK to

Comment (lessw2020) [reply torchtitan/distributed/parallel_dims.py:31]: I did a quick PR to do what Howard is suggesting.  It's pretty similar to the metrics fix I already did for Vblocks.  
It does warn properly and only when needed.
I then tried to update the logging to 'auto fix' the logging for that run but it seems it's too late to update.
https://github.com/pytorch/torchtitan/pull/946
